### PR TITLE
  fix cluster fail

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

- this change to only destroy cluster if `destroy` is true 

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

**Comment by fmoral2** [comment]
this needs to go after https://github.com/rancher/distros-test-framework/pull/116

thats why its failing CI

**Review by rancher-max — DISMISSED** [review]

**Comment by mdrahman-suse** [comment]
Needs rebasing

**Review by rancher-max — COMMENTED** [review]
ah the CI check failed it. I think maybe the best thing to do here is actually move the createcluster logic to a `BeforeSuite`, and remove this destroycluster block altogether. That way, it will run the AfterSuite of destroy if the create cluster fails

**Comment by fmoral2** [comment]
> ah the CI check failed it. I think maybe the best thing to do here is actually move the createcluster logic to a `BeforeSuite`, and remove this destroycluster block altogether. That way, it will run the AfterSuite of destroy if the create cluster fails


- this was failing due to a circular dependency, i have fixed using logger directly

File: pkg/customflag/validate.go [hunk 1]
```diff
@@ -5,30 +5,28 @@ import (
...
 
 	// validate if expected value was sent because it is required for all tests.
 	if expectedValue == "" {
-		shared.LogLevel("error", "expected value was not sent")
+		log.Errorf("expected value was not sent")
 		os.Exit(1)
 	}
 
 	// validate if flag for install version or commit was sent we should have the expected value after upgrade.
 	instalVersionOrCommit := os.Getenv("INSTALL_VERSION_OR_COMMIT")
```
Comment (mdrahman-suse) [inline pkg/customflag/validate.go:26]: typo: `installVersionOrCommit`

File: pkg/customflag/validate.go [hunk 2]
```diff
@@ -44,100 +42,121 @@ func ValidateTemplateFlags() {
...
+
 func validateCiliumTest(expectedValue, valuesUpgrade []string, upgrade string) {
-	ciliumCmdsLenght := 2
+	cmd := os.Getenv("CMD")
+	if cmd != "" {
+		log.Errorf("cmd can not be sent for cilium tests as it is already defined in the test file")
+		os.Exit(1)
+	}
 
+	ciliumCmdsLenght := 2
```
Comment (mdrahman-suse) [inline pkg/customflag/validate.go:89]: typo: `ciliumCmdsLength`

Comment (fmoral2) [reply pkg/customflag/validate.go:89]: thats odd, i swear that we had that English typos check on ci

File: .golangci.yaml [hunk 3]
```diff
@@ -10,9 +10,10 @@ linters:
     - gosimple
     - staticcheck
     - nlreturn
-    - misspell
 
 linters-settings:
+  misspell:
```
Comment (mdrahman-suse) [inline .golangci.yaml:15]: ❤️

File: pkg/customflag/validate.go [hunk 4]
```diff
@@ -44,100 +41,121 @@ func ValidateTemplateFlags() {
...
 
 func validateMultusTest(expectedValue, valuesUpgrade []string, upgrade string) {
-	multusCmdsLenght := 4
+	cmd := os.Getenv("CMD")
+	if cmd != "" {
+		log.Errorf("cmd can not be sent for multus tests as it is already defined in the test file")
+		os.Exit(1)
+	}
 
+	multusCmdsLenght := 4
```
Comment (mdrahman-suse) [inline pkg/customflag/validate.go:110]: typo: `multusCmdsLength`

File: pkg/customflag/validate.go [hunk 5]
```diff
@@ -44,100 +41,121 @@ func ValidateTemplateFlags() {
...
-	product := os.Getenv("ENV_PRODUCT")
-	k3scomponentsCmdsLenght := 10
+	cmd := os.Getenv("CMD")
+	if cmd != "" {
+		log.Errorf("cmd can not be sent for components tests as it is already defined in the test file")
+		os.Exit(1)
+	}
+
+	k3scomponentsCmdsLenght := 9
 	rke2componentsCmdsLenght := 8
```
Comment (mdrahman-suse) [inline pkg/customflag/validate.go:133]: Typos here too for `length`

File: pkg/customflag/validate.go [hunk 6]
```diff
@@ -44,100 +41,121 @@ func ValidateTemplateFlags() {
...
 		if upgrade != "" {
 			if len(valuesUpgrade) != k3scomponentsCmdsLenght {
-				shared.LogLevel("error", "mismatched length commands: %d x expected values upgrade: %d",
+				log.Errorf("mismatched length commands: %d x expected values upgrade: %d",
 					k3scomponentsCmdsLenght, len(valuesUpgrade))
 				os.Exit(1)
 			}
 		}
 	case "rke2":
 		if len(expectedValue) != rke2componentsCmdsLenght {
```
Comment (mdrahman-suse) [inline pkg/customflag/validate.go:151]: Here typo `length`

**Review by mdrahman-suse — APPROVED** [review]

**Comment by fmoral2** [comment]
@mdrahman-suse  sorry i had push a change not supposed to be pushed, i was only testing a cmd for k3s etcd

**Review by mdrahman-suse — APPROVED** [review]

**Review by rancher-max — APPROVED** [review]

**Pull request merged**